app/src/main/java/com/pvp/app/ui/screen/steps/StepViewModel.kt

```diff
@@ -0,0 +1,65 @@
+package com.pvp.app.ui.screen.steps
+
+import androidx.health.connect.client.records.StepsRecord
+import android.os.Build
+import androidx.annotation.RequiresApi
+import androidx.health.connect.client.HealthConnectClient
+import androidx.health.connect.client.PermissionController
+import androidx.health.connect.client.permission.HealthPermission
+import androidx.health.connect.client.records.ExerciseSessionRecord
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.pvp.app.api.HealthConnectService
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import java.time.Instant
+import java.time.LocalDate
+import java.time.ZoneId
+import javax.inject.Inject
+
+@HiltViewModel
+class StepViewModel @Inject constructor(
+    private val healthConnectClient: HealthConnectClient,
+    private val healthConnectService: HealthConnectService
+) : ViewModel() {
+
+    val PERMISSIONS = setOf(
+        HealthPermission.getReadPermission(StepsRecord::class),
+        HealthPermission.getReadPermission(ExerciseSessionRecord::class)
+    )
+
+    private val _stepsCount = MutableStateFlow(0)
+    val stepsCount = _stepsCount.asStateFlow()
+
+    val permissionsLauncher = PermissionController.createRequestPermissionResultContract()
+
+    suspend fun permissionsGranted(): Boolean {
+        val granted = healthConnectClient.permissionController.getGrantedPermissions()
+        return granted.containsAll(PERMISSIONS)
+    }
+
+    @RequiresApi(Build.VERSION_CODES.UPSIDE_DOWN_CAKE)
+    fun updateTodaysSteps() {
+        viewModelScope.launch {
+            val endTime = Instant.now()
+            val startTime = LocalDate.now().atStartOfDay(ZoneId.systemDefault()).toInstant()
+
+            try {
+                val stepsRecords: List<StepsRecord> = healthConnectService.readActivityData(
+                    recordClass = StepsRecord::class,
+                    startTime = startTime,
+                    endTime = endTime
+                )
+
+                val totalSteps = stepsRecords.sumOf { it.count.toInt() }
+
+                _stepsCount.value = totalSteps
+            } catch (e: Exception) {
+
```

Comment:
Redundant empty line.

In what case an exception could be thrown? If there are multiple possible cases and some of them are because of the user or our used service fault, we should inform user about this error.

If we are going to handle this, a toast could be shown. Toast requires a `Context` object, you could get it in the composable with `LocalContext.current` and then call extended function: `showToast`.

Reply:
Error should only be thrown if permissions are not sufficient. For example if the composable would not check for permissions first.

As StepViewModel and StepScreen will most likely not be used and deleted later, I don't think it's necessary to fix this.

In case we decide to keep it (or reuse the updateSteps method in other ViewModel), I will add a comment that the exception should be handled.